src/app/client/src/app/modules/program-dashboard/components/program-datasets/program-datasets.component.ts

```diff
@@ -273,7 +275,7 @@ export class DatasetsComponent implements OnInit, OnDestroy {
       this.reportForm.reset();
       this.reportForm.controls.solution.setValue($event.value);
       this.reportForm.controls.programName.setValue(program);
-
+      this.displayFilters['Resource'] = [$event.source.triggerValue]
```

Comment:
check for undefined case also in all place

Reply:
Whenever the click event on a filter dropdown triggers only then this value will be assigned so there won't be a case of undefined source or triggerValue (as it'd be the cause of the event)

Reply:
if event is not there this will give a undefined issue please use the ? operator

Reply:
Sure will do @rajeevsathish